www/include/home/customViews/triggers/loadServiceFromHost.php

```diff
@@ -62,27 +62,29 @@
 $xml->startElement('response');
 try {
     $xml->startElement('options');
-    if ($data) {
+    if ($hostId) {
         $aclString = $centreon->user->access->queryBuilder(
             'AND',
             's.service_id',
             $centreon->user->access->getServicesString('ID', $monitoringDb)
         );
         $sql = "SELECT service_id, service_description
         		FROM service s, host_service_relation hsr
-        		WHERE hsr.host_host_id = " . $db->escape($data) . "
+        		WHERE hsr.host_host_id = :hostId
         		AND hsr.service_service_id = s.service_id ";
         $sql .= $aclString;
         $sql .= " UNION ";
         $sql .= " SELECT service_id, service_description
         		FROM service s, host_service_relation hsr, hostgroup_relation hgr
         		WHERE hsr.hostgroup_hg_id = hgr.hostgroup_hg_id
-        		AND hgr.host_host_id = ".$db->escape($data)."
+        		AND hgr.host_host_id = :hostId
         		AND hsr.service_service_id = s.service_id ";
         $sql .= $aclString;
         $sql .= " ORDER BY service_description ";
-        $res = $db->query($sql);
-        while ($row = $res->fetchRow()) {
+        $stmt = $db->prepare($sql);
+        $stmt->bindValue(':hostId', $hostId, PDO::PARAM_INT);
+        $stmt->execute();
+        while ($row = $stmt->fetchRow()) {
```

Comment:
```suggestion
        while ($row = $stmt->fetch(\PDO::FETCH_ASSOC)) {
```

Reply:
the fetchRow() method is in a while loop